src/auth.c

```diff
@@ -53,11 +53,17 @@ unsigned int authenticated_since_start = 0;
 
 static void binauth_action(t_client *client, const char *reason)
 {
+	char lockfile[] = "/tmp/ndsctl.lock";
```

Comment:
See issue #409 
and PR #411 

From the PR:

> This solution uses a lock file. The issue may be somewhere in mutex or epoll handling and would probably be better fixed there if that is the case, but this current work around seems reliable.

**_True that PR #411 does not fix the root cause of the problem, but it does reliably prevent the fairly regular total hangups of NDS experienced at busy sites._**

I spent some time in a "modify/test/modify" loop trying to understand why this happens and came up with nothing other than finding that if LOCK_CLIENT_LIST is disabled in the code it makes no difference, implying that LOCK_CLIENT_LIST is not working....

Recently someone found that the deadlock can still occur if ndsctl is called from a binauth script. See issue #503 

This commit here prevents ndsctl from being called from a binauth script - there is no reason to use ndsctl from binauth anyway, but "just in case".

Reply:
> we have to fix this issue on a different way

We _**should.**_ 
I tried.....